ftl/common/ftl_util.py

```diff
@@ -27,3 +27,16 @@ def __enter__(self):
     def __exit__(self, unused_type, unused_value, unused_traceback):
         end = time.time()
         logging.info('%s took %d seconds', self.descriptor, end - self.start)
+
+
+def descriptor_parser(descriptor_files, ctx):
+    descriptor = None
+    for f in descriptor_files:
+        if ctx.Contains(f):
+            descriptor = f
+            descriptor_contents = ctx.GetFile(descriptor)
+            break
+    if not descriptor:
+        logging.info('No package descriptor found. No packages installed.')
+        return [None]
+    return [descriptor_contents]
```

Comment:
Why does this return a list?

Reply:
The descriptor_parser returns a list of inputs for creating packages.  For level 1 this is [list_of_all_packages/entire_descriptor] and for level 2 this looks like [pkg_1, pkg_2, ...]